### PR TITLE
feat: support specify hash algorithm in files.add

### PR DESCRIPTION
**Pull request by alanshaw**
### Specify multihash hashing alogrithm

This allows the optional param `--hash` (CLI) or `?hash=` (HTTP API) to be passed to `files.add` in order to specify the hash algorithm that should be used when generating the multihash(es) for the added content.

Passing this option will set the CID version option to `true`, as per the Go implementation.

### Change `object/get` and `object/data` to accept CIDs

This brings js-ipfs more in line with go-ipfs. Where both `object/get` and `object/data` accept a CID. It's also needed by `js-ipfs-api` when adding files (see further discussion in https://github.com/ipfs/js-ipfs-api/pull/597).

~~**requires https://github.com/ipfs/js-ipfs-unixfs-engine/pull/185**~~
~~**requires https://github.com/ipfs/js-ipfs-api/pull/597**~~

### CODE REVIEW

File: src/core/components/object.js [hunk 1]
```diff
@@ -178,20 +164,19 @@ module.exports = function object (self) {
       }
     }),
 
-    get: promisify((multihash, options, callback) => {
+    get: promisify((hash, options, callback) => {
```
Comment (diasdavid) [inline src/core/components/object.js]: same nitpick as js-ipfs-api, they follow the same interface :)

**Comment by diasdavid** [comment]
@alanshaw do the tests fail to you locally as well?

**Comment by alanshaw** [comment]
Yes these have failed for me too :( I'll take a look asap.

**Comment by diasdavid** [comment]
@alanshaw this needs a rebase from master (things just moved around)

**Comment by diasdavid** [comment]
@alanshaw are you available to finish this PR?

**Comment by diasdavid** [comment]
@hacdias wanna look into this one?

**Comment by diasdavid** [comment]
Done on https://github.com/ipfs/js-ipfs/pull/1308

**Pull request closed without merging**